src/server.ts

```diff
@@ -46,23 +47,42 @@ connection.onInitialize((params: InitializeParams) => {
   return result;
 });
 
+function clearDiagnostics(uri: DocumentUri): void {
+  connection.sendDiagnostics({ uri, diagnostics: [] });
+}
+
+documents.onDidClose((change) => {
+  clearDiagnostics(change.document.uri);
+});
+
+/**
+ * NOTE: Listening on deleted files should NOT be necessary to trigger the clearing of Diagnostics,
+ * since the `documents.onDidClose()` callback should take care of it. However, for some reason,
+ * on automated tests of the SOQL VS Code extension, the 'workbench.action.close*Editor' commands
+ * don't trigger the `onDidClose()` callback on the language server side.
+ *
+ * So, to be safe (and to make tests green) we explicitly clear diagnostics also on deleted files:
+ */
+connection.onDidChangeWatchedFiles((watchedFiles: DidChangeWatchedFilesParams) => {
+  const deletedUris = watchedFiles.changes
+    .filter((change) => change.type === FileChangeType.Deleted)
+    .map((change) => change.uri);
+  deletedUris.forEach(clearDiagnostics);
+});
+
```

Comment:
NOTE: The testing of this functionality is covered by the SOQL VS Code extension